### DISCUSSION TITLE
Add sample lost status support [9000]

### DISCUSSION THREAD

**Discussion by ivanpauno** [eProsima/Fast-DDS#2432] [Ideas]
### Feature request

Currently, there's no support for sample lost status in subscriptions:

https://github.com/eProsima/Fast-DDS/blob/9125239274040384af8b1ad981817333fc4f9d58/include/fastdds/dds/subscriber/DataReader.hpp#L281-L283

Implementing that will unblock a similar feature in ros2: https://github.com/ros2/rmw_fastrtps/issues/395.
The corresponding callback in a `SubscriberListener` will also be needed in order to be able to trigger an event.

**Comment by EduPonz** [comment]
Hi @ivanpauno ,

Sorry for the late reply, we've been out of office. Currently we are focusing on stability, but we will try to allocate some time for this in upcoming development cycles, we'll keep you posted.

**Comment by richiware** [comment]
@ivanpauno This feature is already on the product (see [Pull Request](https://github.com/eProsima/Fast-DDS/pull/2282))

**Reply by ivanpauno** [reply]
Awesome, thanks!
